common/lib/xmodule/xmodule/modulestore/tests/test_search_index.py

```diff
@@ -0,0 +1,276 @@
+"""
+Unit tests for search indexers
+"""
+from lazy.lazy import lazy
+import os
+import mock
+import json
+
+from uuid import uuid4
+from datetime import datetime
+from unittest import TestCase
+
+from opaque_keys.edx.locator import CourseLocator, LibraryLocator, BlockUsageLocator, LibraryUsageLocator
+
+from xmodule.modulestore import ModuleStoreWriteBase
+from xmodule.modulestore.exceptions import ItemNotFoundError
+from xmodule.modulestore.search_index import get_indexer_for_location, CoursewareSearchIndexer, LibrarySearchIndexer
+
+
+class TestIndexerForLocation(TestCase):
+    """ Tests for search indexers factory method """
+    def test_given_course_locator_returns_courseware_indexer(self):
+        locator = mock.Mock(spec=CourseLocator)
+        indexer = get_indexer_for_location(locator)
+        self.assertIsInstance(indexer, CoursewareSearchIndexer)
+
+    def test_given_library_locator_returns_library_indexer(self):
+        locator = mock.Mock(spec=LibraryLocator)
+        indexer = get_indexer_for_location(locator)
+        self.assertIsInstance(indexer, LibrarySearchIndexer)
+
+    def test_given_course_item_locator_returns_courseware_indexer(self):
+        locator = mock.Mock(spec=BlockUsageLocator)
+        locator.course_key = mock.Mock(spec=CourseLocator)
+        indexer = get_indexer_for_location(locator)
+        self.assertIsInstance(indexer, CoursewareSearchIndexer)
+
+    def test_given_library_item_locator_returns_library_indexer(self):
+        locator = mock.Mock(spec=LibraryUsageLocator)
+        locator.course_key = mock.Mock(spec=LibraryLocator)
+        indexer = get_indexer_for_location(locator)
+        self.assertIsInstance(indexer, LibrarySearchIndexer)
+
+
+class SearchIndexerBaseTestMixin(object):
+    """ COmmon features for search indexers tests"""
+    ROOT_ID = None
+    TEST_INDEX_FILENAME = "test_root/index_file.dat"
+    modulestore = None
+
+    def setUp(self):
+        """
+        SetUp method. Sincerely yours, CO.
+        Setting up mock search indexer backing file as prescribed by MOCK_SEARCH_BACKING_FILE setting
+        """
+        super(SearchIndexerBaseTestMixin, self).setUp()
+        # this test implicitly uses MockSearchEngine, which in turn uses settings.MOCK_SEARCH_BACKING_FILE
+        # as backing file - so we create it
+        with open(self.TEST_INDEX_FILENAME, "w+") as index_file:
+            json.dump({}, index_file)
+
+        self.modulestore = mock.Mock(spec=ModuleStoreWriteBase)
+
+    def tearDown(self):
+        """
+        TearDown method. Sincerely yours, CO.
```

Comment:
@Kelketek Captain Obvious. I grew tired of making up docstrings for methods with self-explanatory names, or methods that have special meaning everyone agreed upon long time ago. :) Since this confuses you I'll remove it  :)